### PR TITLE
Simple generic component (module) test

### PR DESCRIPTION
**Pull request by elertan**
This should be relatively easy to understand for people that are starting out with shaku.

### CODE REVIEW

**Comment by Mcat12** [comment]
Thanks for the PR (the first, again)!

You mentioned that it would be helpful to have something like this as an example, so perhaps this code should be moved to the `examples` folder?

File: shaku/tests/simple_generic_component.rs [hunk 1]
```diff
@@ -0,0 +1,60 @@
+//! Modules and services can be generic. Based off of issue #2:
+//! https://github.com/Mcat12/shaku/issues/2
+
+use shaku::{module, Component, Interface, HasComponent};
+
+trait Animal {
+   fn audible_identification(&self) -> &str;
```
Comment (Mcat12) [inline shaku/tests/simple_generic_component.rs:7]: Maybe instead of `identification` use `sound`?

**Comment by elertan** [comment]
Sounds like a good idea.

If I can figure out how to use sqlx Connection effectively I’ll also add a practical example using that

File: shaku/tests/simple_generic_component.rs (hunk 1 above)
Comment (elertan) [reply shaku/tests/simple_generic_component.rs]: Done.

**Comment by elertan** [comment]
I moved the test to an example binary.
I'll try adding a more practical example as well, which uses sqlx's Connection trait.

File: examples/simple_generic_component/Cargo.toml [hunk 2]
```diff
@@ -0,0 +1,12 @@
+[package]
```
Comment (Mcat12) [inline examples/simple_generic_component/Cargo.toml:1]: You can avoid the whole `Cargo.toml` boilerplate by placing the `main.rs` file into `examples` like the other example files: https://doc.rust-lang.org/cargo/guide/project-layout.html

**Pull request merged**